src/components/DataTypes/array.tsx

```diff
@@ -190,25 +143,114 @@ export const getFilter = (schema: JSONSchema, value: string) => {
 		return {
 			minItems: 1,
 			items: {
-				properties: Object.keys(schema.items.properties).reduce(
-					(props: NonNullable<JSONSchema['properties']>, propKey: string) => {
-						props[propKey] = {
-							// @ts-expect-error
-							regexp: {
-								pattern: regexEscape(value),
-								flags: 'i',
+				anyOf: [
+					...Object.entries(schema.items.properties).reduce(
+						(props: any[], [propKey, propValue]: [string, JSONSchema]) => {
+							const subSchema = isFullTextSearch
+								? { ...propValue, oneOf: undefined, enum: undefined }
+								: propValue;
+							const model = getDataModel(subSchema);
+							const operators = model?.operators(schema);
+							const newOperator = convertOperator(operator, operators);
+							const filter = model?.createFilter(
+								key,
+								propKey,
+								newOperator,
+								value,
+								subSchema,
+								true,
+							);
+							props.push(filter);
+							return props;
+						},
+						[],
+					),
+				],
+			},
+		};
+	} else if (
```

Comment:
We don't need this else since we're returning in the if, right?

Reply:
this is one of the exceptions I don't like... That's not the same thing as the one we receive from above. this one will return a contain with a filter, the other one return an items, anyof filter